### PR TITLE
Adds islist() and list2params()

### PR DESCRIPTION
**Pull request by ike709**
Both survived basic testing, but it's probable that the latter has cases I haven't considered.

### CODE REVIEW

File: OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs [hunk 1]
```diff
@@ -550,6 +557,40 @@ static class DreamProcNativeRoot {
...
+        [DreamProc("list2params")]
+        [DreamProcParameter("List")]
+        public static DreamValue NativeProc_list2params(DreamObject instance, DreamObject usr, DreamProcArguments arguments)
+        {
+            if (!arguments.GetArgument(0, "List").TryGetValueAsDreamList(out DreamList list)) throw new ArgumentException("list2params was passed a non-list");
+            
+            StringBuilder paramBuilder = new StringBuilder();
+            
+            if (list.IsAssociative()) {
+                Dictionary<DreamValue, DreamValue> associativeValues = list.GetAssociativeValues();
```
Comment (ZeWaka) [inline OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs:570]: This looks like it has the same issue as #92.

**Review by wixoaGit — CHANGES_REQUESTED** [review]
The keys and values need to be URL encoded using `HttpUtility.UrlEncode`

Currently `list2params(list("@"="<"))` returns `@=<` when it should return `%40=%3c`

File: OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs [hunk 2]
```diff
@@ -550,6 +557,40 @@ static class DreamProcNativeRoot {
 
             throw new Exception("Cannot check length of " + value + "");
         }
+        
+        [DreamProc("list2params")]
+        [DreamProcParameter("List")]
+        public static DreamValue NativeProc_list2params(DreamObject instance, DreamObject usr, DreamProcArguments arguments)
+        {
+            if (!arguments.GetArgument(0, "List").TryGetValueAsDreamList(out DreamList list)) throw new ArgumentException("list2params was passed a non-list");
```
Comment (wixoaGit) [inline OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs:565]: Return an empty string instead of throwing an exception.

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs [hunk 3]
```diff
@@ -550,6 +557,40 @@ static class DreamProcNativeRoot {
...
+                        paramBuilder.Append('&');
+                    }
+                }
+            } else {
+                foreach (DreamValue val in values)
+                {
+                    paramBuilder.Append(val.Value);
+                    paramBuilder.Append('&');
+                }
+            }
```
Comment (wixoaGit) [inline OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs:588]:
```suggestion
            Dictionary<DreamValue, DreamValue> associativeValues = list.GetAssociativeValues();
            foreach (DreamValue entry in values) {
                if (associativeValues.ContainsKey(entry))
                {
                    paramBuilder.Append($"{entry.Value}={associativeValues[entry].Value}");
                    paramBuilder.Append('&');
                } else {
                    paramBuilder.Append(entry.Value);
                    paramBuilder.Append('&');
                }
            }
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Comment by ike709** [comment]
Addressed.

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**